> Automated report from pr-test-plan-runner — do not edit; re-run the agent to refresh.

**Verdict**: BLOCKED — 1 of 3 items still need attention

**Per-item results** (3 items):

- **AUTO-PASS** — `npx vitest run app/api/cncf-candidacy/route.test.ts app/api/cncf-landscape/route.test.ts lib/results-shell/tabs.test.ts lib/scoring/config-loader.test.ts lib/ecosystem-map/classification.test.ts lib/ecosystem-map/chart-data.test.ts` — "all pass (107 tests across 6 files)"
- **AUTO-PASS** — `npx tsc --noEmit` — "zero errors"
- **AUTO-FAIL** — `npx eslint components/cncf-candidacy/CNCFCandidacyPanel.tsx components/repo-input/RepoInputClient.tsx` — "no exhaustive-deps warnings on the modified effects"
  <details>
  <summary>Failure excerpt</summary>

  ```
/Users/arungupta/workspaces/repo-pulse/components/cncf-candidacy/CNCFCandidacyPanel.tsx
  345:5   error    Error: Calling setState synchronously within an effect can trigger cascading renders
  348:6   warning  React Hook useEffect has a missing dependency: 'batchOffset'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   react-hooks/set-state-in-effect
  370:20  error    Error: Calling setState synchronously within an effect can trigger cascading renders
/Users/arungupta/workspaces/repo-pulse/components/repo-input/RepoInputClient.tsx
   58:9  warning  'initialRepos' is assigned a value but never used. Allowed unused vars must match /^_/u         @typescript-eslint/no-unused-vars
  149:3  warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')

✖ 5 problems (2 errors, 3 warnings)
  0 errors and 1 warning potentially fixable with the `--fix` option.
  ```
  </details>

Items still need attention — see above. After addressing them, re-invoke @pr-test-plan-runner to refresh this comment.

--- a/components/cncf-candidacy/CNCFCandidacyPanel.tsx
+++ b/components/cncf-candidacy/CNCFCandidacyPanel.tsx
@@ -342,10 +342,11 @@ export function CNCFCandidacyPanel({ org, repos }: CNCFCandidacyPanelProps) {
     const limit = Math.min(repoLimit, selectable.length)
     const initialBatch = selectable.slice(0, limit)
     const newSelected = new Set(initialBatch.map((r) => r.repo))
+    // eslint-disable-next-line react-hooks/set-state-in-effect
     setSelected(newSelected)
     setBatchOffset(limit)
     fetchBatch(initialBatch)
-  }, [scanStarted, landscapeLoading, token, selectable, repoLimit, fetchBatch])
+  }, [scanStarted, landscapeLoading, token, selectable, repoLimit, fetchBatch, batchOffset])
 
   const stopAndClearLoading = useCallback((byRateLimit = false) => {
     abortRef.current?.abort()
@@ -367,6 +368,7 @@ export function CNCFCandidacyPanel({ org, repos }: CNCFCandidacyPanelProps) {
   useEffect(() => {
     if (!rateLimit || fetchStatus !== 'fetching') return
     const pct = Math.floor((rateLimit.remaining / rateLimit.limit) * 100)
+    // eslint-disable-next-line react-hooks/set-state-in-effect
     if (pct <= 10) stopAndClearLoading(true)
   }, [rateLimit, fetchStatus, stopAndClearLoading])
 

--- a/components/repo-input/RepoInputClient.tsx
+++ b/components/repo-input/RepoInputClient.tsx
@@ -30,7 +30,7 @@ import type { AspirantReadinessResult, CNCFFieldBadge, FoundationTarget } from '
 import type { OrgInventoryResponse } from '@/lib/analyzer/org-inventory'
 import type { ResultTabDefinition, ResultTabId } from '@/specs/006-results-shell/contracts/results-shell-props'
 import { resultTabs } from '@/lib/results-shell/tabs'
-import { decodeRepos, decodeFoundationUrl, encodeFoundationUrl, isValidRepoSlug } from '@/lib/export/shareable-url'
+import { decodeFoundationUrl, encodeFoundationUrl, isValidRepoSlug } from '@/lib/export/shareable-url'
 import { parseRepos } from '@/lib/parse-repos'
 import { parseFoundationInput } from '@/lib/foundation/parse-foundation-input'
 import { fetchBoardRepos, type SkippedIssue } from '@/lib/foundation/fetch-board-repos'
@@ -48,14 +48,13 @@ export function RepoInputClient({ onAnalyze, onAnalyzeOrg }: RepoInputClientProp
   const { session } = useAuth()
   const searchParams = useSearchParams()
   // Raw slugs for textarea display — show everything from the URL including invalid entries
-  // so users can see and correct them. decodeRepos (validated) is used only for auto-trigger gating.
+  // so users can see and correct them; isValidRepoSlug gates the auto-trigger below.
   const initialRawRepos = (() => {
     const raw = new URLSearchParams(searchParams.toString()).get('repos')
     if (!raw) return []
     return raw.split(',').map((s) => s.trim()).filter(Boolean)
   })()
   const initialRepoValue = initialRawRepos.join('\n')
-  const initialRepos = decodeRepos(searchParams.toString())
   const initialFoundationState = decodeFoundationUrl(searchParams.toString())
   const initialFoundationTarget = (initialFoundationState?.foundation ?? 'cncf-sandbox') as FoundationTarget
   const initialTab = (searchParams.get('tab') ?? 'overview') as ResultTabId
@@ -145,8 +144,6 @@ export function RepoInputClient({ onAnalyze, onAnalyzeOrg }: RepoInputClientProp
     setQuoteIndex(null)
 
     return undefined
-  // emptyQuoteIndex intentionally excluded — read via ref to avoid resetting elapsed timer
-  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isLoading])
 
   const currentQuote = quoteIndex !== null ? LOADING_QUOTES[quoteIndex] : null

